From d7fed5639e21e7e72b0f8e89d10b5b5ce90ab98a Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Mon, 23 Oct 2017 12:45:45 -0400
Subject: [PATCH] rcu: fix implicit include for ULONG_CMP_LT

We see this on an -rt variant of the kernel, but it does not seem
that there is anything really -rt specific other than unmasking
an implicit include dependency as follows:

kernel/rcu/rcu_segcblist.c: In function 'rcu_segcblist_advance':
kernel/rcu/rcu_segcblist.c:396:3: error: implicit declaration of function 'ULONG_CMP_LT' [-Werror=implicit-function-
declaration]

Add the include explicitly to ensure it is not an issue in any
future kernels.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
---
 kernel/rcu/rcu_segcblist.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/rcu/rcu_segcblist.c b/kernel/rcu/rcu_segcblist.c
index 2b62a38b080f..73c0534ecb3e 100644
--- a/kernel/rcu/rcu_segcblist.c
+++ b/kernel/rcu/rcu_segcblist.c
@@ -22,6 +22,7 @@
 
 #include <linux/types.h>
 #include <linux/kernel.h>
+#include <linux/rcupdate.h>
 #include <linux/interrupt.h>
 
 #include "rcu_segcblist.h"
-- 
2.14.1

